fixed log message levels

fametrano opened the pull request:
Downgraded some LOG.info to LOG.debug

Upgraded the LOG.info in trezorlib/transport/hid.py to LOG.warning (similarly to trezorlib/transport/webusb.py)

Re-enabled the LOG.debug that was commented out in trezorlib/device.py

See: https://github.com/bitcoin-core/HWI/pull/372

prusnak commented:
Please fix the Travis failure by running `black`

matejcik commented:
please explain why you want any of that changed.

I see no point in returning the "computer-generated entropy" message, as the data is visible in protobuf debug.

HID not present is definitely not worth a warning, as we now only need HID to communicate with freshly-unboxed T1 without any firmware updates - a minority usecase, to be sure?

per-transport enumeration seems right at the `info` level.

the only reasonable change is dropping `info` to `debug` in TrezorClient instantiation, but in that case I'd argue maybe the log item should be dropped altogether?

fametrano commented:
> Please fix the Travis failure by running `black` on `python/src/trezorlib/{client,device}.py`
> 
> : https://travis-ci.org/github/trezor/trezor-firmware/builds/719026288#L381

done.

> I see no point in returning the "computer-generated entropy" message, as the data is visible in protobuf debug.

removed as suggested

> HID not present is definitely not worth a warning, as we now only need HID to communicate with freshly-unboxed T1 without any firmware updates - a minority usecase, to be sure?

I understand and reverted that change

> the only reasonable change is dropping `info` to `debug` in TrezorClient instantiation, but in that case I'd argue maybe the log item should be dropped altogether?

that change is kept in this PR

> per-transport enumeration seems right at the `info` level.

actually that change is the driving reason for this PR. I use trezorlib (through HWI) in other projects and every time devices are enumerated the per-transport enumeration LOG.info message "pollutes" the end system LOG.
As they say "your mileage may vary", but a final user's LOG.info should not be polluted by third party library.

matejcik commented:
> final user's LOG.info should not be polluted by third party library.

Are you implying that libraries should never use `LOG.info` at all -- but are free to use `debug`, `warning` and `error`? That seems like a weird argument. Is there some sort of widespread consensus about this?

If not, this should be solved in your application via the following snippet:
```
logging.getLogger("trezorlib").setLevel(logging.ERROR)  # or WARNING or whichever you want
```

matejcik commented:
actually maybe you should not be setting level on the root logger if you want `INFO` messages from your app but not from your dependencies?

fametrano commented:
thank you for the setLevel suggestion.

Anyway, I stand by my suggestion: could you imagine if the python standard libraries would log INFO ?
For a third party library is acceptable to log CRITICAL, ERROR, and WARNING; not so much to log INFO

prusnak commented:
>  For a third party library is acceptable to log CRITICAL, ERROR, and WARNING; not so much to log INFO

Is this documented somewhere or that's something you came up with?

fametrano commented:
> Is this documented somewhere or that's something you came up with?

it is not documented as far as I know, but it is my experience that I have never seen my LOG polluted with INFO from standard libraries (not even by third party libraries, but trezorlib). Is your experience different?

Of course, it goes without saying that I have no problem in maintaining my own fork with this changeset or use setLevel if this PR is controversial. No holy war, especially considering that I would just lose it :-)

matejcik commented:
> Anyway, I stand by my suggestion: could you imagine if the python standard libraries would log INFO ?
> For a third party library is acceptable to log CRITICAL, ERROR, and WARNING; not so much to log INFO

i can absolutely imagine that, yes :)

I believe my original `setLevel` suggestion was misguided though: what you *should* do is enable logging for your own namespace only.

I.e., I presume you have something like this in your code?
```
root_logger = logging.getLogger()
root_logger.addHandler(...)
root_logger.setLevel(logging.INFO)
```
which sets the log level for the full module hierarchy, including standard libraries.

This is in fact what I would want to do if I wanted to see INFO output of third parties.

instead your code should be configuring its own logger only:
```
my_logger = logging.getLogger(__name__)
...
```
see https://github.com/trezor/trezor-firmware/blob/9a330f34756c7f553c187d007ddce14eea73ca65/python/src/trezorlib/log.py#L61-L65 -- in the `enable_debug_output` helper, we are only enabling log messages for `trezorlib` logger (and its children), so no 3rd party messages will come out.

---

closing, as I believe that this should be solved in your code rather than in all your dependencies one-by-one
